### PR TITLE
[forward port from 2.12.x] GitHub Actions: build and test on Windows

### PR DESCRIPTION
**Pull request by SethTisue**
and remove obsolete CI scripts

forward-ports #9496 and #9585

### CODE REVIEW

**Comment by SethTisue** [comment]
before merging, we should consider reducing the overlap with #9579

in particular, it would be nice to not just have one big monolithic log, which is what `testAll` gives us

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # Note that we don't use olafurpg/setup-scala; it wouldn't buy us anything
+      # over setup-java. (We don't want csbt or xsbt; we prefer the standard
+      # sbt launch script, which comes preinstalled on Windows (and Ubuntu).)
+      - name: Setup Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: adopt
```
Comment (griggt) [inline .github/workflows/ci.yml:28]: Oh, the new setup-java action now supports AdoptOpenJDK.  Nice!  I didn't catch that change on #9496.

**Comment by SethTisue** [comment]
This PR achieves parity with what we had on Jenkins and what we have on 2.12.x. Therefore, I suggest we merge it.

(But yes, we should eventually reduce the overlap with #9579. It would be nice to not just have one big monolithic log, which is what `testAll` gives us.)

**Comment by lrytz** [comment]
The main reason for splitting up `testAll` is that travis runs the two parts concurrently – I think not only to have results faster, but also to avoid timeouts. Do we have concurrent jobs/stages on GHA? What is the job timeout limit?

**Comment by lrytz** [comment]
The `scripts/jobs/integrate/ide` script still shows up on jenkins, e.g. https://scala-ci.typesafe.com/job/scala-2.13.x-validate-main/14599/parameters. Not sure if removing it will cause failures?

**Review by lrytz — APPROVED** [review]

**Review by dwijnand — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -3,6 +3,7 @@ name: Scala Merge CI
 on:
   push:
     branches: ['2.*.x']
+  pull_request:   # TODO remove before merging PR
```
Comment (dwijnand) [inline .github/workflows/ci.yml]: remove

**Comment by SethTisue** [comment]
> The main reason for splitting up testAll is that travis runs the two parts concurrently – I think not only to have results faster, but also to avoid timeouts. Do we have concurrent jobs/stages on GHA? What is the job timeout limit?

Beats me. Let's keep these questions in mind as we continue to evolve these scripts. I have set a reminder to look in on the Windows runs every few days and see how they're doing. (As I have long done with the corresponding Jenkins jobs.)

> The scripts/jobs/integrate/ide script still shows up on jenkins, e.g. scala-ci.typesafe.com/job/scala-2.13.x-validate-main/14599/parameters. Not sure if removing it will cause failures?

We haven't seen failures on 2.12.x, but regardless, I'll certainly keep an eye out.

But also, I have just removed the references from the Jenkins job configs.

**Comment by griggt** [comment]
>  Do we have concurrent jobs/stages on GHA? 

This is possible, yes.

> What is the job timeout limit?

6 hours per job, 72 hours for the workflow as a whole.

https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration#usage-limits

I don't believe there are any timeouts based on output as Travis has.

**Comment by dwijnand** [comment]
> > The main reason for splitting up testAll is that travis runs the two parts concurrently – I think not only to have results faster, but also to avoid timeouts. Do we have concurrent jobs/stages on GHA? What is the job timeout limit?
> 
> Beats me. Let's keep these questions in mind as we continue to evolve these scripts. I have set a reminder to look in on the Windows runs every few days and see how they're doing. (As I have long done with the corresponding Jenkins jobs.)

Yeah, avoiding timeouts so forks could pass on Travis.  And then if there has to be two parts, split them as equally as possible to complete fastest.

**Pull request merged**